core/services/vrf/v2/listener_v2.go

```diff
@@ -519,68 +530,80 @@ func (lsn *listenerV2) processPendingVRFRequests(ctx context.Context) {
 // MaybeSubtractReservedLink figures out how much LINK is reserved for other VRF requests that
 // have not been fully confirmed yet on-chain, and subtracts that from the given startBalance,
 // and returns that value if there are no errors.
-func MaybeSubtractReservedLink(q pg.Q, startBalance *big.Int, chainID uint64, subID *big.Int, vrfVersion vrfcommon.Version) (*big.Int, error) {
-	var (
-		reservedLink string
-		query        string
-	)
+func (lsn *listenerV2) MaybeSubtractReservedLink(ctx context.Context, startBalance *big.Int, chainID *big.Int, subID *big.Int, vrfVersion vrfcommon.Version) (*big.Int, error) {
+	var metaField string
 	if vrfVersion == vrfcommon.V2Plus {
-		query = V2PlusReservedLinkQuery
+		metaField = GlobalSubId
 	} else if vrfVersion == vrfcommon.V2 {
-		query = V2ReservedLinkQuery
+		metaField = SubId
 	} else {
 		return nil, errors.Errorf("unsupported vrf version %s", vrfVersion)
 	}
 
-	err := q.Get(&reservedLink, query, chainID, subID.String())
+	txes, err := lsn.chain.TxManager().FindTxesByMetaFieldAndStates(ctx, metaField, subID.String(), reserveEthLinkQueryStates, chainID)
 	if err != nil && !errors.Is(err, sql.ErrNoRows) {
-		return nil, errors.Wrap(err, "getting reserved LINK")
+		return nil, errors.Wrap(err, "TXM FindTxesByMetaFieldAndStates failed")
 	}
 
-	if reservedLink != "" {
-		reservedLinkInt, success := big.NewInt(0).SetString(reservedLink, 10)
-		if !success {
-			return nil, fmt.Errorf("converting reserved LINK %s", reservedLink)
+	reservedLinkSum := big.NewInt(0)
+	// Aggregate non-null MaxLink from all txes returned
+	for _, tx := range txes {
+		var meta *txmgrtypes.TxMeta[common.Address, common.Hash]
+		meta, err = tx.GetMeta()
```

Comment:
Just checking, but this doesn't error out if the meta is just null for whatever reason (not sure if that's possible) right?

Reply:
It does not. It returns nil if meta is nil. It only errors if there is an issue unmarshaling the json object. I believe with the nature of the query meta wouldn't be nil but I've added a check for sanity